ha/core/cluster/cluster_manager.py

```diff
@@ -296,3 +304,152 @@ def controller_list(self) -> list:
             [list]: list of controllers.
         """
         return list(self._controllers.keys())
+
+    def get_system_health(self, element: CLUSTER_ELEMENTS = CLUSTER_ELEMENTS.CLUSTER.value, depth: int = 1, **kwargs) -> dict:
+        """
+        Return health status for the requested elements.
+
+        Args:
+            element ([CLUSTER_ELEMENTS]): The element whose health status is to be returned.
+            depth ([int]): A depth of elements starting from the input "element" that the health status
+                is to be returned.
+            **kwargs([dict]): Variable number of arguments that are used as filters,
+                e.g. "id" of the input "element".
+
+        Returns:
+            ([dict]): Returns dictionary. {"status": "Succeeded"/"Failed"/"Partial", "output": "", "error": ""}
+                status: Succeeded, Failed, Partial
+                output: Dictionary with element health status
+                error: Error information if the request "Failed"
+        """
+
+        try:
+            # Currently, only "id" supported in the variable arguments.
+            unsupported_element = True
+            for supported_element in CLUSTER_ELEMENTS:
+                if element == supported_element.value:
+                    unsupported_element = False
+                    break
+
+            # TODO: Fix below temporary code
+            if element != CLUSTER_ELEMENTS.NODE.value:
+                unsupported_element = True
+
+            if unsupported_element:
+                return {"status": const.STATUSES.FAILED.value, "output": "", "error": "Invalid element"}
+            if kwargs and GET_SYS_HEALTH_ARGS.ID.value not in kwargs:
+                return {"status": const.STATUSES.FAILED.value, "output": "", "error": "Invalid filter argument(s)"}
+
+            element_id = HEALTH_STATUSES.UNKNOWN.value
+            if kwargs:
+                element_id = kwargs["id"]
+
+            # Fetch the health status
+            system_health_controller = SystemHealthController(self._confstore)
+            self._status_dict = system_health_controller.get_status(CLUSTER_ELEMENTS.CLUSTER.value)
+
+            # Remove any keys which are not for the health status.
+            ignore_keys = []
+            for key in self._status_dict:
+                if "health" not in key:
+                    ignore_keys.append(key)
+            for key in ignore_keys:
+                del self._status_dict[key]
+
+            # Get the requested element level in the health hierarchy
+            self._health_hierarchy = HealthHierarchy()
+            element_level = self._health_hierarchy.get_element_level(element)
+
+            # TODO: Fix below
```

Comment:
Remove / fix the TODO

Reply:
Yes, I will fix this in next PR for the same task. The task will not be closed till then.

Reply:
okay